src/Camera.ts

```diff
@@ -18,37 +18,33 @@ export class Camera {
     this.minWorldPos = minWorldPos
     this.worldDimensions = worldDimensions
 
-    this.shakeStartTime = 0
-    this.shakeLastMoveTime = 0
+    this.ttl = 0
+    this.cooldownTtl = 0
     this.shakeOffset = vec2.fromValues(0, 0)
   }
 
   shake(): void {
-    this.shakeStartTime = Date.now()
-    this.shakeLastMoveTime = 0
+    this.ttl = 0.25
+    this.cooldownTtl = 0
   }
 
-  update(): void {
-    if (this.shakeStartTime === 0) {
-      return
-    }
-
-    const now = Date.now()
-
-    if (this.shakeStartTime < now - 250) {
+  update(dt: number): void {
+    this.ttl -= dt
+    if (this.ttl <= 0) {
+      this.ttl = 0
       vec2.zero(this.shakeOffset)
-      this.shakeStartTime = 0
       return
     }
 
-    if (this.shakeLastMoveTime < now - 1000 / 60) {
+    this.cooldownTtl -= dt
+    if (this.cooldownTtl <= 0) {
+      this.cooldownTtl += 1 / 60
```

Comment:
The TTL units are seconds, so this amounts to one change in the shake offset per 60fps frame.